clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -53,6 +54,9 @@
     /* the list of topics the user has requested */
     private final Set<String> subscription;
 
+    /* the list of topic partitions the user has unsubscribed from but not committed their offsets yet */
+    private final Map<TopicPartition, TopicPartitionState> pendingUnsubscribe;
```

Comment:
Is it possible to use a flag in `TopicPartitionState` instead of a whole new collection?